dDatabase/CDClientDatabase/CDClientTables/CDRewardCodesTable.cpp

```diff
@@ -0,0 +1,53 @@
+#include "CDRewardCodesTable.h"
+
+void CDRewardCodesTable::LoadValuesFromDatabase() {
+
+	// First, get the size of the table
+	unsigned int size = 0;
+	auto tableSize = CDClientDatabase::ExecuteQuery("SELECT COUNT(*) FROM RewardCodes");
+	while (!tableSize.eof()) {
+		size = tableSize.getIntField(0, 0);
+
+		tableSize.nextRow();
+	}
+
+	tableSize.finalize();
```

Comment:
Can be removed, RAII for cdclient resultsets automatically do this with a try catch(...)